commands/project/dev/index.ts

```diff
@@ -50,15 +46,61 @@ export async function handler(
     process.exit(EXIT_CODES.ERROR);
   }
 
+  let targetAccountId = providedAccountId;
+
+  let profile: HsProfileFile | undefined;
+
+  if (!targetAccountId && useV3Api(projectConfig.platformVersion)) {
+    if (args.profile) {
+      logProfileHeader(args.profile);
+
+      profile = loadProfile(projectConfig, projectDir, args.profile);
+
+      if (!profile) {
+        uiLine();
+        process.exit(EXIT_CODES.ERROR);
+      }
+
+      targetAccountId = profile.accountId;
+
+      logProfileFooter(profile);
+    } else {
+      // A profile must be specified if this project has profiles configured
+      await exitIfUsingProfiles(projectConfig, projectDir);
```

Comment:
For all of the commands that leverage profiles. We will exit if you have profiles configured AND you don't specify a profile for the command. It has to be explicitly provided